src/main/java/pl/agh/cs/io/Rule.java

```diff
@@ -21,10 +22,15 @@ public Rule(String path) {
     }
 
     public void handle(WindowState state) {
+
         if (prevState == WindowState.FOREGROUND || prevState == WindowState.BACKGROUND) {
             if (prevState != state && prevState != WindowState.CLOSED) {
                 createNewTime(prevState);
             }
+
+            if (restriction != null) {
```

Comment:
Może warto podawać restriction w konstruktorze, jeśli zawsze musi być.

Reply:
Domyślnie aplikacja jest monitorowana ale nie jest ograniczana, dopiero później można nałożyć ograniczenie

Reply:
Z tego co widzę, to chyba nie zawsze musi być, ale z nullami jest brzydko, więc może warto by było zrobić jakieś EmptyRestriction, które nic nie robi. Ale może to trochę czepianie się

Reply:
Zmieniłem na Optional

Reply:
Spoko ja już wcześniej dałem approve także chyba można mergować